ray-operator/controllers/ray/common/pod.go

```diff
@@ -160,6 +187,40 @@ func BuildPod(podTemplateSpec v1.PodTemplateSpec, rayNodeType rayiov1alpha1.RayN
 
 	setContainerEnvVars(&pod.Spec.Containers[rayContainerIndex], rayNodeType, rayStartParams, svcName)
 
+	// add health check for head node
+
+	// health check only works for 1.13
+	if major, minor, _, err := ParseVersion(rayVersion); err == nil && major*100+minor >= 113 {
+		if rayNodeType == rayiov1alpha1.HeadNode {
+			probe := &v1.Probe{}
+			if pod.Spec.Containers[rayContainerIndex].ReadinessProbe == nil {
+				pod.Spec.Containers[rayContainerIndex].ReadinessProbe = probe
+			}
+			probe = pod.Spec.Containers[rayContainerIndex].ReadinessProbe
+			if probe.Exec == nil && probe.HTTPGet == nil && probe.TCPSocket == nil {
+				probe.Exec = &v1.ExecAction{Command: []string{
+					"python", "-c",
+					"from ray._private.gcs_utils import check_health;import sys;" +
+						fmt.Sprintf("sys.exit(not check_health('localhost:%d'))",
+							DefaultRedisPort),
+				}}
+			}
+			if probe.InitialDelaySeconds == 0 {
+				probe.InitialDelaySeconds = 10
+			}
+			if probe.FailureThreshold == 0 {
+				// This number is set to 20 to make default tests happy
```

Comment:
Could you elaborate on this?
It looks like we need to fail for 200 seconds straight (about 3 minutes) before any recovery logic kicks in.
Is that what we want? cc @iycheng @scv119 @brucez-anyscale

Reply:
yeah, is 20 too hight?

Reply:
I set it to 10 for now

Reply:
it is updated. now it is configurable and also the default value can be configured.